packages/preferences/src/browser/preference-tree-model.ts

```diff
@@ -107,10 +107,14 @@ export class PreferenceTreeModel extends TreeModelImpl {
                 this.updateFilteredRows(PreferenceFilterChangeSource.Scope);
             }),
             this.filterInput.onFilterChanged(newSearchTerm => {
+                const wasFiltered = this._isFiltered;
                 this.lastSearchedLiteral = newSearchTerm;
                 this.lastSearchedFuzzy = newSearchTerm.replace(/\s/g, '');
                 this._isFiltered = newSearchTerm.length > 2;
                 this.updateFilteredRows(PreferenceFilterChangeSource.Search);
+                if (!wasFiltered && this.isFiltered) {
+                    this.expandAll();
```

Comment:
Makes node expansion the default when searching. At the moment, it only occurs when you switch modes unfiltered -> filtered, but it could be applied on every change of the filter, if that's preferable.

Reply:
That feels much nicer. I think having the expansion applied on every change of the filter would be a good addition, seems that VSCode does it that way as well